File: python/native_historical_export.py
from client import create_gateway_client
import compassiot.compass.v1.time_pb2 as time
import compassiot.native.v1.native_pb2 as native


def main():
    client = create_gateway_client()

    request = native.HistoricalExportRequest(
        filter_vins=[],
        filter_type=native.HistoricalExportFilterType.TRIP,
        filter_dates=time.DateTimeRange(
            start=time.LocalDate(
                day=1,
                month=10,
                year=2023
            ),
            end=time.LocalDate(
                day=10,
                month=10,
                year=2023
            ),
            day_of_week=[
                time.DayOfWeek.MONDAY,
                time.DayOfWeek.WEDNESDAY,
                time.DayOfWeek.FRIDAY,
                time.DayOfWeek.SUNDAY
            ],
            hour_of_day=[i for i in range(24)]
        )
    )

    response = client.NativeHistoricalExport(request)

    print(response.csv_links)

     
if __name__ == "__main__":
    main()

File: typescript/src/native_historical_export.ts
import * as time from "@buf/compassiot_model.bufbuild_es/compassiot/compass/v1/time_pb"
import * as native from "@buf/compassiot_api.bufbuild_es/compassiot/native/v1/native_pb"

import client from "./client"

const request = new native.HistoricalExportRequest({
  filterVins: [],
  filterDates: new time.DateTimeRange({
    start: new time.LocalDate({
      day: 1,
      month: 10,
      year: 2023
    }),
    end: new time.LocalDate({
      day: 31,
      month: 10,
      year: 2023
    }),
    dayOfWeek: [
      time.DayOfWeek.SATURDAY,
      time.DayOfWeek.SUNDAY
    ],
    hourOfDay: Array.from(Array(24).keys())  // 0 to 23
  })
})

const response = await client.nativeHistoricalExport(request)

console.log(response.toJsonString({ prettySpaces: 2 }))
